packages/webpack-module-federation/src/useAreRemotesReady.ts

```diff
@@ -0,0 +1,31 @@
+import { useEffect, useState } from "react";
+
+import { registrationStatus } from "./registerRemoteModules.ts";
+
+export interface UseAreRemotesReadyOptions {
+    // The interval is in milliseconds.
+    interval?: number;
+}
+
+export function useAreRemotesReady({ interval = 10 }: UseAreRemotesReadyOptions = {}) {
+    // Using a state hook to force a rerender once ready.
+    const [, isReady] = useState(false);
```

Comment:
it really feels weird to read isReady here instead of setIsReady. 

at first look, it seems like an error